src/restruct_poc/beams.hpp

```diff
@@ -0,0 +1,609 @@
+#pragma once
+
+#include <array>
+#include <numeric>
+
+#include <KokkosBlas.hpp>
+
+#include "beams_calcs.hpp"
+#include "interpolation.h"
+#include "types.hpp"
+
+#include "src/gebt_poc/quadrature.h"
+
+namespace oturb {
+
+//------------------------------------------------------------------------------
+// Beam input data structures
+//------------------------------------------------------------------------------
```

Comment:
The input data structures should be in their own file rather.

Reply:
I agree, feel free to move them or I'll take care of it at some point.